sqlglot/optimizer/simplify.py

```diff
@@ -430,6 +432,74 @@ def simplify_parens(expression):
     return expression
 
 
+CONSTANTS = (
+    exp.Literal,
+    exp.Boolean,
+    exp.Null,
+)
+
+
+def simplify_coalesce(expression):
+    # COALESCE(x) -> x
+    if (
+        isinstance(expression, exp.Coalesce)
+        and not expression.expressions
+        # COALESCE is also used as a Spark partitioning hint
+        and not isinstance(expression.parent, exp.Hint)
+    ):
+        return expression.this
+
+    if not isinstance(expression, COMPARISONS):
+        return expression
+
+    if isinstance(expression.left, exp.Coalesce):
+        coalesce = expression.left
+        other = expression.right
+    elif isinstance(expression.right, exp.Coalesce):
+        coalesce = expression.right
+        other = expression.left
+    else:
+        return expression
+
+    # COALESCE(x, y, z)
+    #                ^ grab that guy
+    coalesce_args = coalesce.expressions
+    if coalesce_args:
+        rightmost_arg = coalesce_args[-1]
+    else:
+        return expression
+
+    # COALESCE(x, 1) = 2
+    # becomes
+    # (
+    #     NOT COALESCE(x) IS NULL
+    #     AND COALESCE(x) = 2
+    # )
+    # OR (
+    #     COALESCE(x) IS NULL
+    #     AND 1 = 2
+    # )
+    # This may seem more complex, but subsequent simplify steps will further reduce this.
+    #
+    # This transformation is valid for non-constants,
+    # but I think it really only does anything if they are both constants.
+    if isinstance(rightmost_arg, CONSTANTS) and isinstance(other, CONSTANTS):
+        rightmost_arg.pop()
+
+        return exp.or_(
+            exp.and_(
+                coalesce.copy().is_(exp.null()).not_(),
+                expression.copy(),
```

Comment:
same with this one, and_ is copy by default, unless you want to do and_ copy=False